lib/core/presentation/pages/event_tickets/event_buy_tickets_page/sub_pages/event_tickets_summary_page/handler/wait_for_payment_notification_handler.dart

```diff
@@ -1,37 +1,99 @@
 import 'dart:async';
 
-import 'package:app/core/presentation/widgets/common/dialog/lemon_alert_dialog.dart';
-import 'package:app/i18n/i18n.g.dart';
-import 'package:app/theme/typo.dart';
-import 'package:auto_route/auto_route.dart';
+import 'package:app/core/domain/payment/entities/payment.dart';
+import 'package:app/core/domain/payment/input/get_payment_input/get_payment_input.dart';
+import 'package:app/core/domain/payment/payment_enums.dart';
+import 'package:app/core/domain/payment/payment_repository.dart';
+import 'package:app/core/domain/web3/web3_repository.dart';
+import 'package:app/core/utils/web3_utils.dart';
+import 'package:app/injection/register_module.dart';
 import 'package:flutter/material.dart';
 
 class WaitForPaymentNotificationHandler {
   Timer? timer;
 
-  start(BuildContext context) {
-    timer = Timer(const Duration(minutes: 3), () {
-      final t = Translations.of(context);
-      final colorScheme = Theme.of(context).colorScheme;
-
-      showDialog(
-        barrierDismissible: false,
-        context: context,
-        builder: (context) {
-          return LemonAlertDialog(
-            buttonLabel: t.common.actions.ok,
-            child: Text(
-              t.event.eventPayment.paymentNotificationTimeoutMessage,
-              style: Typo.medium.copyWith(
-                color: colorScheme.onSecondary,
-              ),
-            ),
-            onClose: () async {
-              AutoRouter.of(context).root.popUntilRouteWithPath('/events');
-            },
-          );
-        },
+  Future<Payment?> _checkPayment(String paymentId) async {
+    int remainingAttempt = 10;
+    Payment? payment;
+    while (remainingAttempt > 0) {
+      Future.delayed(
+        const Duration(
+          seconds: 5,
+        ),
       );
+      final paymentResult = await getIt<PaymentRepository>().getPayment(
+        input: GetPaymentInput(
+          id: paymentId,
+        ),
+      );
+      payment = paymentResult.fold(
+        (l) => null,
+        (r) => r,
+      );
+      if (payment?.state == PaymentState.succeeded) {
+        return payment;
+      }
+      remainingAttempt--;
+    }
+    return payment;
+  }
+
+  start(
+    BuildContext context, {
+    required String paymentId,
+    Function()? onPaymentDone,
+    Function()? onPaymentFailed,
+  }) async {
+    timer = Timer(const Duration(minutes: 3), () async {
```

Comment:
Similar to the 3 minutes, recommend define some default like DEFAULT_DURATION_CHECK_PAYMENT